sklearn/neighbors/_classification.py

```diff
@@ -278,6 +292,29 @@ def predict_proba(self, X):
             by lexicographic order.
         """
         if self.weights == "uniform":
+            # TODO: systematize this mapping of metric for
+            # PairwiseDistancesReductions.
+            metric = (
+                "euclidean"
+                if self.metric == "minkowski" and self.p == 2
+                else self.metric
+            )
```

Comment:
This could be achieved with a small private helper function `_canonical_metric` or something to avoid repeating this code snippet.

Reply:
Added a helper function `_adjusted_metric` to `_classification.py` for now, though eventually perhaps this should be resolved directly in the dispatchers to ensure consistent behavior (currently only a problem due to KNC's unique choice of using `metric="minkowski"` as its default).